front/src/applications/operationalStudies/views/Scenario/components/Timetable/Timetable.tsx

```diff
@@ -94,6 +212,43 @@ const Timetable = ({
     [selectedTimetableItemIds]
   );
 
+  const sortTrainScheduleSets = useCallback(
+    (set1: TrainScheduleSet, set2: TrainScheduleSet): number => {
+      const set1CatalogueId = set1.catalogue_entry_id;
+      const set2CatalogueId = set2.catalogue_entry_id;
+
+      const catalogueName1 = set1CatalogueId
+        ? (catalogueEntrySample.find((entry) => entry.id === set1CatalogueId)?.name ?? null)
+        : null;
+      const catalogueName2 = set2CatalogueId
+        ? (catalogueEntrySample.find((entry) => entry.id === set2CatalogueId)?.name ?? null)
+        : null;
+
+      // If no catalogue name, put at the end
+      if (!catalogueName1 && catalogueName2) return 1;
+      if (catalogueName1 && !catalogueName2) return -1;
+
+      // If sandbox, put at the very end
+      if (set1.is_sandbox && !set2.is_sandbox) return 1;
+      if (!set1.is_sandbox && set2.is_sandbox) return -1;
+
+      // If both have catalogue names, build full names and compare
+      if (catalogueName1 && catalogueName2) {
+        const fullName1 = `${catalogueName1} | ${set1.name ?? ''}`;
```

Comment:
We also use the same "string format" in the `TrainScheduleSetTab` component at line 74 (so in another file).
And both are related, here we sort the array with the label which  is displayed over there. 
It can becomes out of sync in the futur, maybre we can  create a toString function ?